hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/XceiverServerRatis.java

```diff
@@ -686,4 +693,17 @@ void notifyGroupRemove(RaftGroupId gid) {
   void notifyGroupAdd(RaftGroupId gid) {
     raftGids.add(gid);
```

Comment:
lets add an entry to  leaderIdMap on groupAdd and remove it on groupRemove, else we can leak memory.